src/biogeophys/HillslopeHydrologyMod.F90

```diff
@@ -0,0 +1,1140 @@
+                   patch%wtgcell(pc) = sum_wtgrc
+
+                else
+                   write(iulog,*) 'no pft in column ',c, col%hill_pftndx(c)
+                   write(iulog,*) 'pfts ',c,patch%itype(col%patchi(c):col%patchf(c))
+                   write(iulog,*) 'weights ',c,patch%wtcol(col%patchi(c):col%patchf(c))
+                   write(iulog,*) 'location ',c,grc%londeg(col%gridcell(c)),grc%latdeg(col%gridcell(c))
+                   
+                endif
+             endif
+          enddo    ! end loop c
+       enddo ! end loop l
+    enddo    ! end loop nc
+    !$OMP END PARALLEL DO
+
+  end subroutine HillslopePftFromFile
+
+  !-----------------------------------------------------------------------
+  subroutine HillslopeStreamOutflow(bounds, &
+       waterstatebulk_inst, waterfluxbulk_inst,streamflow_method)
+    !
+    ! !DESCRIPTION:
+    ! Calculate discharge from stream channel
+    !
+    ! !USES:
+    use LandunitType    , only : lun                
+    use GridcellType    , only : grc                
+    use ColumnType      , only : col                
+    use WaterFluxBulkType   , only : waterfluxbulk_type
+    use WaterStateBulkType  , only : waterstatebulk_type
+    use spmdMod         , only : masterproc
+    use clm_varcon      , only : spval, ispval, grlnd 
+    use landunit_varcon , only : istsoil
+    use ncdio_pio
+    use clm_time_manager , only : get_step_size_real
+     
+    !
+    ! !ARGUMENTS:
+    type(bounds_type), intent(in) :: bounds
+    integer,  intent(in)  :: streamflow_method
+    type(waterstatebulk_type), intent(inout) :: waterstatebulk_inst
+    type(waterfluxbulk_type),  intent(inout) :: waterfluxbulk_inst
+
+    integer               :: c, l, g, i, j
+    integer               :: nstep
+    real(r8) :: dtime                                   ! land model time step (sec)
+    real(r8)              :: cross_sectional_area
+    real(r8)              :: stream_depth
+    real(r8)              :: hydraulic_radius
+    real(r8)              :: flow_velocity
+    real(r8)              :: overbank_area
+!    real(r8), parameter   :: manning_roughness = 0.05
+    real(r8), parameter   :: manning_roughness = 0.03
+    real(r8), parameter   :: manning_exponent  = 0.667
+
+    character(len=*), parameter :: subname = 'HillslopeStreamOutflow'
+
+    !-----------------------------------------------------------------------
+    associate(                                                            & 
+         stream_water_volume     =>    waterstatebulk_inst%stream_water_lun            , & ! Input:  [real(r8) (:)   ] stream water volume (m3)
+         qstreamflow             =>    waterfluxbulk_inst%qstreamflow_lun               &  ! Input:  [real(r8) (:)   ] stream water discharge (m3/s)
+         )
+
+      ! Get time step
+      dtime = get_step_size_real()
+
+      do l = bounds%begl,bounds%endl
+         qstreamflow(l) = 0._r8
+         if(lun%itype(l) == istsoil) then
```

Comment:
- [ ] This should also check lun%active(l) to avoid doing calculations for inactive landunits, which can lead to trying to do calculations on garbage values. (In practice this won't matter in this specific case, but it's good to stay consistent about this.)